rust/main/submitter/src/chain_tx_adapter/chains/builder.rs

```diff
@@ -0,0 +1,30 @@
+// TODO: re-enable clippy warnings
+#![allow(dead_code)]
+
+use hyperlane_base::settings::ChainConf;
+use hyperlane_core::{HyperlaneDomain, HyperlaneDomainProtocol};
+
+use crate::chain_tx_adapter::{
+    chains::{
+        cosmos::CosmosChainTxAdapter, ethereum::EthereumChainTxAdapter,
+        sealevel::SealevelChainTxAdapter,
+    },
+    AdaptsChain,
+};
+
+pub struct ChainTxAdapterBuilder {}
+
+impl ChainTxAdapterBuilder {
+    pub fn build(conf: &ChainConf) -> Box<dyn AdaptsChain> {
+        use HyperlaneDomainProtocol::*;
+
+        let adapter: Box<dyn AdaptsChain> = match conf.domain.domain_protocol() {
+            Ethereum => Box::new(EthereumChainTxAdapter::new(conf.clone())),
+            Fuel => todo!(),
+            Sealevel => Box::new(SealevelChainTxAdapter::new(conf.clone())),
```

Comment:
Ideally, we should transform generic `ChainConf` here into chain-specific struct here.